Support merging attributes from "attrs"

Dresdn opened the pull request:
My attempt at merging `attrs`. Let me know if this is what you had envisioned for https://github.com/wrabit/django-cotton/discussions/189

Can update the docs too as this opens up some powerful use cases, including fully supporting [Widget Attributes](https://docs.djangoproject.com/en/5.2/ref/forms/widgets/#widget).

Dresdn committed:
Support merging attributes from "attrs"

wrabit commented:
Hi @Dresdn thanks for the PR. Will this also work for the case we just want to proxy the `{{ attrs }}` down to another component, i.e. `:attrs="attrs"`?

Dresdn commented:
Yep! I saw somewhere (issue or discussion) you mentioned `{{ atts }}` just returns a dict, so works the same way. I did test this, but can add a test to the suite if it helps.

wrabit commented:
@Dresdn great, an additional test for that would be good if you don't mind ensuring it covers multiple attribute types (string, dynamic, boolean, dynamic component (:is)

Dresdn committed:
Add test for attrs proxying with multiple data types

cadolphs commented:
:+1: for this PR. Let me know if there's any way I can help. Super excited both for the merging and for the passing down.

My current use-case example: Have a `base.html` version of a UI element and then different variations of it, all of which should pass their `attrs` to the `base` as needed.

wrabit commented:
@Dresdn thanks for the work on this. Would you mind drafting document changes as you see it?

Dresdn committed:
add docs for :attrs dynamic attribute usage

Dresdn commented:
@wrabit - let me know what you think about the docs changes.

I added a `Components - 5.1` section under `{{ attrs }}` as it seemed to make sense to explain what the "special dynamic attribute :attrs" does. Additionally, I added a new `Usage Patterns` example and was originally going to use a whole `Form Widget` to `<input />` example, decided just to stick a generic "higher order component" language to keep it a little agnostic.

wrabit commented:
Looks good. I will merge the docs build fix and add a note about `:attrs="attrs"` use case also, thanks for the PR!
